ergotree-interpreter/src/sigma_protocol/prover.rs

```diff
@@ -434,7 +434,7 @@ fn polish_simulated<P: Prover + ?Sized>(
                         for (idx, kid) in unproven_children.clone().enumerated() {
                             if kid.is_real() {
                                 count_of_real += 1;
-                                if count_of_real >= ct.k {
+                                if count_of_real > ct.k {
```

Comment:
An off-by-one issue. Note that the associated code on scala side increments count_of_real after the conditional check(https://github.com/ScorexFoundation/sigmastate-interpreter/blob/bd1906e3a0a4f93039a357c8d3ca4343b135dcf2/sigmastate/src/main/scala/sigmastate/interpreter/ProverInterpreter.scala#L249). So this is equivalent.

Reply:
I found this issue because the following assertion was failing: https://github.com/ergoplatform/sigma-rust/blob/b922854061bbb9a01391091128045806505b1332/ergotree-interpreter/src/sigma_protocol/sig_serializer.rs#L97-L100

Reply:
Oh wow! Great job!